mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -102,6 +113,35 @@ public boolean setOperation(Operation operation) {
 		return true;
 	}
 
+	public boolean isCliQuery() {
+		return isCliQuery;
+	}
+
+	public boolean setCliQuery(boolean isCliQuery) {
+		this.isCliQuery = isCliQuery;
+		return true;
+	}
+
+	public boolean setInvalidJsonInPrefix(String invalidJsonInPrefix) {
```

Comment:
I don't understand why you called it `inPrefix`?

Reply:
I changed the definition of `JsonObject` rule to catch the rest of the input in this field if the essential part fails when the `JsonObject` rule is called. The rest of the input is not a valid JSON but I don't know what exactly is wrong - this field simply stores everything after position of a supposed JSON object. If we could "fix" this JSON part of the input it would be in the prefix part of the captured string. This was my reasoning for naming but I see it is confusing. Do you think I should rename it?